### PR TITLE
Request to fix (#117), user cannot sign out during create_profile sec…

### PR DESCRIPTION
**Pull request by ericpb**
Request to fix (#117), user cannot sign out during create_profile section

## Summary by CodeRabbit

- **New Features**
	- Enhanced the sign-out process by implementing a secure POST request instead of a GET request, improving user experience and security.
	- Introduced a server-side action for signing out users, ensuring proper session management and redirection after logout.

- **Bug Fixes**
	- Resolved issues related to the previous sign-out implementation, ensuring it adheres to best practices for state-changing operations.

### CODE REVIEW

File: src/routes/(admin)/account/create_profile/+page.svelte [hunk 1]
```diff
@@ -113,7 +113,9 @@
       <div class="text-sm text-slate-800 mt-14">
         You are logged in as {session?.user?.email}.
         <br />
-        <a class="underline" href="/account/sign_out"> Sign out </a>
+        <form action="/account/sign_out" method="POST">
+          <button type="submit" class="underline">Sign out</button>
+        </form>
```
[the remark this answers is not in the document]
Comment (scosman) [reply src/routes/(admin)/account/create_profile/+page.svelte:118]: We can just use the sign out link format we use elsewhere: 

`<a href="/account/sign_out" class="link">Sign Out</a>`

No need to add a new endpoint. 

POST does make sense, but don't want to fragment it and have a mix of methods across the template.

Sounds good?

**Review by scosman — CHANGES_REQUESTED** [review]

File: src/routes/(admin)/account/sign_out/+page.server.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { redirect } from "@sveltejs/kit"
```
Comment (scosman) [inline src/routes/(admin)/account/sign_out/+page.server.ts:1]: Can just delete this change (see above comment)

File: src/routes/(admin)/account/create_profile/+page.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericpb) [reply src/routes/(admin)/account/create_profile/+page.svelte:118]: I understand the preference for consistency. However the current approach of using <a href="/account/sign_out"> is not working on the create_profile page. I think due to the differences in how SvelteKit handles page navigation between layouts and individual pages.
I'm quite new to SvelteKit and this template so I could be wrong; happy to discuss further.

**Comment by scosman** [comment]
Thanks for the bug report @ericpb 

You were totally right. Made a smaller fix here to let the standard sign out path work: https://github.com/CriticalMoments/CMSaasStarter/commit/f280d590afbc0b58cd9d573db8a200b3a9c472e6

**Pull request closed without merging**